src/components/views/messages/SenderProfile.js

```diff
@@ -42,22 +42,23 @@ export default function SenderProfile(props) {
             : null,
     ];
 
-    let content = '';
-
+    let content;
     if(props.text) {
-        // Replace senderName, and wrap surrounding text in spans with the right class
-        content = _tJsx(props.text, /^(.*)\%\(senderName\)s(.*)$/m, (p1, p2) => [
-            p1 ? <span className='mx_SenderProfile_aux'>{ p1 }</span> : null,
-            nameElem,
-            p2 ? <span className='mx_SenderProfile_aux'>{ p2 }</span> : null,
-        ]);
+        content = _t(props.text, { senderName: () => nameElem });
     } else {
-        content = nameElem;
+        // There is nothing to translate here, so call substitute() instead
+        content = substitute('%(senderName)s', { senderName: () => nameElem });
     }
 
     return (
         <div className="mx_SenderProfile" dir="auto" onClick={props.onClick}>
-            { content }
+            { content.props.children[0] ?
```

Comment:
This is a little bit sad, especially because we're only doing this so that Flair can appear with a different opacity to the sender profile text and yet we don't show Flair (currently) when the aux text is actually used. 

One way to do this is to not use opacity as a cheap way to get the text to be the colour we want. This feels out-of-scope, but maybe leave a "`// XXX:`" comment explaining this.